### PR TITLE
JENKINS-73971  Extract event handlers in NestedViewsSearch/search-results

### PR DESCRIPTION
**Pull request by judovana**
https://issues.jenkins.io/browse/JENKINS-73971

### CODE REVIEW

File: src/main/resources/hudson/plugins/nested_view/NestedViewsSearch/searchscripts.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+        // <![CDATA[
```
Comment (basil) [inline src/main/resources/hudson/plugins/nested_view/NestedViewsSearch/searchscripts.js:1]: CDATA is not needed here. or on line 16.

**Review by basil — COMMENTED** [review]
This is really great, as it gets us halfway to the finish line. To complete this task, we need to remove the `onclick=` HTML attribute, and instead set it programmatically from `searchscripts.js`. To do that, you'll need to use either `document.addEventListener('DOMContentLoaded', …)` or `Behaviour#specify` (as described in https://www.jenkins.io/doc/developer/security/csp/#inline-event-handlers) to write some code that runs when the page loads. That code should find the `<a>` elements whose `onclick` handlers were removed (e.g. through `document.querySelector` and iterating by class name or ID), then add a new `onclick` handler with `element.addListener("click", ...)`. For an example, see:

https://github.com/jenkinsci/scriptler-plugin/pull/120

Once that is done and the result is tested, I am happy to approve this PR.

**Comment by judovana** [comment]
> This is really great, as it gets us halfway to the finish line. To complete this task, we need to remove the `onclick=` HTML attribute, and instead set it programmatically from `searchscripts.js`. To do that, you'll need to use either `document.addEventListener('DOMContentLoaded', …)` or `Behaviour#specify` (as described in https://www.jenkins.io/doc/developer/security/csp/#inline-event-handlers) to write some code that runs when the page loads. That code should find the `<a>` elements whose `onclick` handlers were removed (e.g. through `document.querySelector` and iterating by class name or ID), then add a new `onclick` handler with `element.addListener("click", ...)`. For an example, see:
> 
> [jenkinsci/scriptler-plugin#120](https://github.com/jenkinsci/scriptler-plugin/pull/120)
> 
> Once that is done and the result is tested, I am happy to approve this PR.

Ah sure. Will do.

File: src/main/resources/hudson/plugins/nested_view/NestedViewsSearch/searchscripts.js (hunk 1 above)
Comment (judovana) [reply src/main/resources/hudson/plugins/nested_view/NestedViewsSearch/searchscripts.js:1]: Unless you insists, I will keep it there.  I like when code is somehow wrapped, and on contrary I do nt ot like  the JS's  `myNameSapce = function () {` notation... Anyway.. Yah my bad habit:(

**Comment by judovana** [comment]
@basil  done sir!

**Comment by judovana** [comment]
This is quite educational PR. TY!

**Review by basil — APPROVED** [review]
Thanks for the PR! I trust you can confirm this is all working as expected by manually testing the UI.

**Comment by judovana** [comment]
yup, I did. In adition unittests did its job susprisngly well. ty!

**Comment by basil** [comment]
Thank you very much!

**Pull request merged**